crates/matrix-sdk-base/src/media.rs

```diff
@@ -64,6 +64,46 @@ impl UniqueKey for MediaThumbnailSize {
     }
 }
 
+/// The desired settings of a media thumbnail.
+#[derive(Clone, Debug)]
+pub struct MediaThumbnailSettings {
+    /// The desired size of the thumbnail.
+    pub size: MediaThumbnailSize,
+
+    /// If we want to request an animated thumbnail from the homeserver.
+    ///
+    /// If it is `Some(true)`, the server should return an animated thumbnail if
+    /// the media supports it. If it is `Some(false)`, the server must never
+    /// return an animated thumbnail. If it is `None`, the server should not
+    /// return an animated thumbnail.
+    pub animated: Option<bool>,
```

Comment:
It seems there's no difference between `Some(false)` and `None`; do you confirm this, or is the difference subtle (in which case the code comment should elaborate on that)? If there's no difference, can we use a plain `bool` here instead, set to false by default?

Reply:
The subtle difference is that in one case it's a MUST NOT for the homeserver and in the other case it's a SHOULD NOT. I am not sure if that difference makes sense for clients, it might make more sense to always send `true` or `false`.

Reply:
In theory homeservers should ignore unknown query parameters.

Reply:
Done